data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorConfiguration.java

```diff
@@ -110,12 +118,35 @@ public GenericRegexMatchOperator regexNotEqualOperator() {
 
     @Bean
     public GenericEqualOperator equalOperator() {
-        return new GenericEqualOperator(DataPrepperExpressionParser.EQUAL, equals);
+        final BiPredicate<Object, Object> floatEquals = (lhs, rhs) -> ((Number)lhs).floatValue() == ((Number) rhs).floatValue();
+
+        final Map<Class<?>, Map<Class<?>, BiPredicate<Object, Object>>> equalStrategy = new HashMap<>();
+
+        final Map<Class<?>, BiPredicate<Object, Object>> intOperations = new HashMap<>();
+        intOperations.put(Integer.class, (lhs, rhs) -> (int) lhs == (int) rhs);
+        intOperations.put(Float.class, floatEquals);
+
+        final Map<Class<?>, BiPredicate<Object, Object>> floatOperations = new HashMap<>();
+        floatOperations.put(Integer.class, floatEquals);
```

Comment:
Should this be `floatOperations.put(Integer.class, (lhs, rhs) -> (int) lhs == (int) rhs);`?

Reply:
In this case, the `lhs` is of type `Float`. casting to an int case will cause precision loss. [used here](https://github.com/sbayer55/data-prepper/blob/d7e659ca01ab7cd2c50bd8473d27e385addee152/data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorConfiguration.java#L134)